build_tools/cmake/iree_native_test.cmake

```diff
@@ -89,6 +92,25 @@ function(iree_native_test)
     list(APPEND _RULE_DATA "${_RULE_TEST_INPUT_FILE_ARG}")
   endif()
 
+  # Detect file location with `{{}}` and handle its portability for all entries
+  # in `_RULE_ARGS`.
+  foreach(_ARG ${_RULE_ARGS})
+    string(REGEX MATCH ".+{{(.+)}}" _FILE_ARG "${_ARG}")
+    if(_FILE_ARG)
+      set(_FILE_PATH ${CMAKE_MATCH_1})
```

Comment:
Seems like there could be multiple file paths in a single argument, though it doesn't seem common. I'm not really clear on whether `REGEX MATCH_ALL` is the right way to do that here. I think doing a `MATCH_ALL` without a capture would give you all the matches

```
string(REGEX MATCH_ALL "{{.+}}" _FILES "${_ARG}")
string(REGEX REPLACE "{{(.+)}}" "\\1" _ARG "${_ARG}")

foreach(_FILE IN LISTS _FILES)
  string(REGEX REPLACE "{{(.+)}}" "\\1" _FILE "${_FILE}")
  # adding to data stuff...
endforeach()
```